src/NATS.Client.Core/INatsSerializer.cs

```diff
@@ -18,6 +18,21 @@ public interface ICountableBufferWriter : IBufferWriter<byte>
     int WrittenCount { get; }
 }
 
+public readonly struct FixedSizeMemoryOwner : IMemoryOwner<byte>
```

Comment:
Out of curiosity, what is the benefit of using this over `MemoryPool<byte>.Shared` ?

Reply:
MemoryPool<byte>.Shared.Rent returns *at least* the size asked for, which usually rounds up to some 2^n. This wrapper chops the memory to exact size so when it's passed to stream read/write or down the line to publish or something you won't need to pass the actual size.

Reply:
btw, we are still renting it from MemoryPool.

Reply:
> btw, we are still renting it from MemoryPool.

Ah, I see that now.  Clever!